features/steps/advance_search.py

```diff
@@ -99,3 +109,51 @@ def step_impl(context):
     assert field_value == "NEW"
     field_value = advanced_search_page.get_field_value_from_flawlist("impact")
     assert field_value == "LOW"
+
+
+@then('I am able to search flaws with empty or nonempty value')
+def step_impl(context):
+    advanced_search_page = AdvancedSearchPage(context.browser)
+    for field in EMPTY_FIELDS:
+        advanced_search_page.selectKeyList.click_button()
+        advanced_search_page.selectKeyList.select_element_by_value(field)
+
+        # Check search result with empty value
+        advanced_search_page.click_btn("emptyBtn")
+        advanced_search_page.click_btn("searchBtn")
+        advanced_search_page.first_flaw_exist()
+        if field in ['cve_id', 'owner']:
+            value = advanced_search_page.get_field_value_from_flawlist(field)
+            is_correct = not value.startswith('CVE-') if field == 'cve_id' else value == ''
+            assert is_correct is True
+        else:
+            # cve_description, cvss_scores__score, mitigation, statement
+            advanced_search_page.go_to_first_flaw_detail()
+            time.sleep(1)
+            flaw_page = FlawDetailPage(context.browser)
+            assert flaw_page.get_cvssV3_score() == ''
+            assert flaw_page.get_document_text_field('description') == ''
+            assert flaw_page.get_document_text_field('mitigation') == ''
+            assert flaw_page.get_document_text_field('statement') == ''
+            # Back to advanced search page
+            go_to_advanced_search_page(context.browser)
+            advanced_search_page.selectKeyList.click_button()
+            advanced_search_page.selectKeyList.select_element_by_value(field)
+
+        # Check search result with nonempty value
+        advanced_search_page.click_btn("nonemptyBtn")
+        advanced_search_page.click_btn("searchBtn")
+        advanced_search_page.first_flaw_exist()
+        if field in ['cve_id', 'owner']:
+            value = advanced_search_page.get_field_value_from_flawlist(field)
+            is_correct = value.startswith('CVE-') if field == 'cve_id' else value != ''
+            assert is_correct is True
+        else:
+            advanced_search_page.go_to_first_flaw_detail()
+            time.sleep(1)
+            flaw_page = FlawDetailPage(context.browser)
+            assert flaw_page.get_cvssV3_score() != ''
```

Comment:
Feature: Flaw advance search testing # features/advance_search.feature:1

  Background: Before run scenarios  # features/advance_search.feature:3

  Scenario: Search flaws with empty or nonempty value           # features/advance_search.feature:23
    Given I am an analyst AND I am logged into OSIM             # features/steps/common_steps.py:14 265.476s
    * I set the bugzilla api key and jira api key               # features/steps/common_steps.py:19 8.999s
    Given I go to the advanced search page                      # features/steps/common_steps.py:41 0.098s
    Then I am able to search flaws with empty or nonempty value # features/steps/advance_search.py:114 9.738s
      Traceback (most recent call last):
        File "/Users/axuan/PycharmProjects/osim/venv/lib/python3.9/site-packages/behave/model.py", line 1329, in run
          match.run(runner.context)
        File "/Users/axuan/PycharmProjects/osim/venv/lib/python3.9/site-packages/behave/matchers.py", line 98, in run
          self.func(context, *args, **kwargs)
        File "features/steps/advance_search.py", line 155, in step_impl
          assert flaw_page.get_cvssV3_score() != ''
        File "/Users/axuan/PycharmProjects/osim/features/pages/flaw_detail_page.py", line 434, in get_cvssV3_score
          return find_element_in_page_factory(self, "cvssV3Score").text
        File "/Users/axuan/PycharmProjects/osim/features/page_factory_utils.py", line 23, in find_element_in_page_factory
          element = page_factory_obj.driver.find_element(
        File "/Users/axuan/PycharmProjects/osim/venv/lib/python3.9/site-packages/selenium/webdriver/remote/webdriver.py", line 748, in find_element
          return self.execute(Command.FIND_ELEMENT, {"using": by, "value": value})["value"]
        File "/Users/axuan/PycharmProjects/osim/venv/lib/python3.9/site-packages/selenium/webdriver/remote/webdriver.py", line 354, in execute
          self.error_handler.check_response(response)
        File "/Users/axuan/PycharmProjects/osim/venv/lib/python3.9/site-packages/selenium/webdriver/remote/errorhandler.py", line 229, in check_response
          raise exception_class(message, screen, stacktrace)
      selenium.common.exceptions.NoSuchElementException: Message: Unable to locate element: //span[@class='osim-cvss-score']; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
      Stacktrace:
      RemoteError@chrome://remote/content/shared/RemoteError.sys.mjs:8:8
      WebDriverError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:193:5
      NoSuchElementError@chrome://remote/content/shared/webdriver/Errors.sys.mjs:511:5
      dom.find/</<@chrome://remote/content/shared/DOM.sys.mjs:136:16



Failing scenarios:
  features/advance_search.feature:23  Search flaws with empty or nonempty value